qutebrowser/browser/network/networkmanager.py

```diff
@@ -241,12 +241,21 @@ def clear_rejected_ssl_errors(self, url):
     @pyqtSlot('QNetworkReply', 'QAuthenticator')
     def on_authentication_required(self, reply, authenticator):
         """Called when a website needs authentication."""
-        answer = self._ask("Username ({}):".format(authenticator.realm()),
+        import netrc
+        net = netrc.netrc()
+        authenticators = net.authenticators(reply.url().host())
+        user, password = None, None
+
+        if authenticators:
+            (user, account, password) = authenticators
```

Comment:
The docs say:

> If the netrc file did not contain an entry for the given host, return the tuple associated with the ‘default’ entry.

I'm not sure what that means exactly - but it seems if the user has such a default entry, any website with HTTP authentication they visit will instantly tell them that the credentials are invalid... not sure if that's desirable...